sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ddl.scala

```diff
@@ -34,11 +34,16 @@ import org.apache.spark.sql.types._
  * @param tableDesc the metadata of the table to be created.
  * @param mode the data writing mode
  * @param query an optional logical plan representing data to write into the created table.
+ * @param useExternalSerde whether to use external serde to write data, e.g., Hive Serde. Currently
```

Comment:
This is too hacky. We should not leak hive specific knowledge to general logical plans.

Reply:
This is because all rules related to conversion to data source are located in `RelationConversions`.

If we loose this requirement, we can avoid this flag and let `CreateHiveTableAsSelectCommand` decide to convert it to data source or not.

Reply:
Do you think it is better to put all this conversion stuff of Hive CTAS into `CreateHiveTableAsSelectCommand`?

Reply:
I don't have a clear idea now, but `CreateTable` is a general logical plan for CREATE TABLE, we may even public in to data source/catalog APIs in the future, we should not put hive specific concept here.